actionpack/test/dispatch/feature_policy_test.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require "abstract_unit"
+
+class FeaturePolicyTest < ActiveSupport::TestCase
+  def setup
+    @policy = ActionDispatch::FeaturePolicy.new
+  end
+
+  def test_mappings
+    @policy.midi :self
+    assert_equal "midi 'self'", @policy.build
+
+    @policy.midi :none
+    assert_equal "midi 'none'", @policy.build
+  end
+
+  def test_multiple_sources_for_a_single_directive
+    @policy.geolocation :self, "https://example.com"
+    assert_equal "geolocation 'self' https://example.com", @policy.build
+  end
+
+  def test_single_directive_for_multiple_directives
+    @policy.geolocation :self
+    @policy.usb :none
+    assert_equal "geolocation 'self'; usb 'none'", @policy.build
+  end
+
+  def test_multiple_directives_for_multiple_directives
+    @policy.geolocation :self, "https://example.com"
+    @policy.usb :none, "https://example.com"
+    assert_equal "geolocation 'self' https://example.com; usb 'none' https://example.com", @policy.build
+  end
+
+  def test_invalid_directive_source
+    exception = assert_raises(ArgumentError) do
+      @policy.vr [:non_existent]
+    end
+
+    assert_equal "Invalid HTTP feature policy source: [:non_existent]", exception.message
+  end
+end
+
+class FeaturePolicyIntegrationTest < ActionDispatch::IntegrationTest
+  class PolicyController < ActionController::Base
+    feature_policy only: :index do |f|
+      f.gyroscope :none
+    end
+
+    feature_policy only: :sample_controller do |f|
+      f.gyroscope nil
```

Comment:
I'm not sure how I feel about this API. The idea is that you define a global policy you need to be explicit about overriding it with `nil` to remove it. Thoughts?